src/plugins/dashboard/public/application/actions/clone_panel_action.test.tsx

```diff
@@ -0,0 +1,155 @@
+    ContactCardEmbeddableInput,
+    ContactCardEmbeddableOutput,
+    ContactCardEmbeddable
+  >(CONTACT_CARD_EMBEDDABLE, {
+    firstName: 'Kibana',
+  });
+
+  if (isErrorEmbeddable(contactCardEmbeddable)) {
+    throw new Error('Failed to create embeddable');
+  } else {
+    embeddable = contactCardEmbeddable;
+  }
+});
+
+test('Clone adds a new embeddable', async () => {
+  const dashboard = embeddable.getRoot() as IContainer;
+  const originalPanelCount = Object.keys(dashboard.getInput().panels).length;
+  const originalPanelKeySet = new Set(Object.keys(dashboard.getInput().panels));
+  const action = new ClonePanelAction(coreStart);
+  await action.execute({ embeddable });
+  expect(Object.keys(container.getInput().panels).length).toEqual(originalPanelCount + 1);
+  const newPanelId = Object.keys(container.getInput().panels).find(
+    key => !originalPanelKeySet.has(key)
+  );
+  expect(newPanelId).toBeDefined();
+  const newPanel = container.getInput().panels[newPanelId!];
+  expect(newPanel.type).toEqual(embeddable.type);
+});
+
+test('Clones an embeddable without a saved object ID', async () => {
+  const dashboard = embeddable.getRoot() as IContainer;
+  const panel = dashboard.getInput().panels[embeddable.id] as DashboardPanelState;
+  const action = new ClonePanelAction(coreStart);
+  // @ts-ignore
+  const newPanel = await action.cloneEmbeddable(panel, embeddable.type);
+  expect(newPanel.type).toEqual(embeddable.type);
+});
+
+test('Clones an embeddable with a saved object ID', async () => {
+  const dashboard = embeddable.getRoot() as IContainer;
+  const panel = dashboard.getInput().panels[embeddable.id] as DashboardPanelState;
+  panel.explicitInput.savedObjectId = 'holySavedObjectBatman';
+  const action = new ClonePanelAction(coreStart);
+  // @ts-ignore
+  const newPanel = await action.cloneEmbeddable(panel, embeddable.type);
+  expect(coreStart.savedObjects.client.get).toHaveBeenCalledTimes(1);
+  expect(coreStart.savedObjects.client.find).toHaveBeenCalledTimes(1);
+  expect(coreStart.savedObjects.client.create).toHaveBeenCalledTimes(1);
+  expect(newPanel.type).toEqual(embeddable.type);
+});
+
+test('Gets a unique title ', async () => {
+  coreStart.savedObjects.client.find = jest.fn().mockImplementation(({ search }) => {
+    if (search === '"testFirstTitle"') return { total: 1 };
+    else if (search === '"testSecondTitle"') return { total: 41 };
+    else if (search === '"testThirdTitle"') return { total: 90 };
+  });
+  const action = new ClonePanelAction(coreStart);
+  // @ts-ignore
+  expect(await action.getUniqueTitle('testFirstTitle', embeddable.type)).toEqual(
+    'testFirstTitle (copy)'
+  );
+  // @ts-ignore
+  expect(await action.getUniqueTitle('testSecondTitle (copy 39)', embeddable.type)).toEqual(
+    'testSecondTitle (copy 40)'
+  );
+  // @ts-ignore
+  expect(await action.getUniqueTitle('testSecondTitle (copy 20)', embeddable.type)).toEqual(
+    'testSecondTitle (copy 40)'
```

Comment:
Not sure I understand this - shouldn't the next title be `copy 41`?

Reply:
The titling uses a system like file copies in windows, the order looks like this: 

OG Title
OG Title (copy)
OG Title (copy 1)

That means that the next title is always the number of similar titles - 1. So if there were 41 similar titles, the next title would be OG Title (Copy 40)